packages/cohere/src/cohere-chat-language-model.ts

```diff
@@ -218,6 +238,60 @@ export class CohereChatLanguageModel implements LanguageModelV1 {
                 return;
               }
 
+              case 'tool-calls-chunk': {
+                if (value.tool_call_delta) {
+                  if (value.tool_call_delta.index != toolCalls.length - 1) {
```

Comment:
what if an earlier index comes again? shouldn't we check `toolCall[index] != null` or something like that?

Reply:
this has been updated to create a new tool call only after checking if the element at said index is empty

also added a test to check for out of order stream deltas [here](https://github.com/vercel/ai/pull/2658/files#diff-437429bac0dbc9688ae63e95f835bc93d1f2a3735948e800fa3192f1718d991dR643)